ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,518 @@
+
+            /// \brief Returns an operation place by an operation name following framework
+            /// conventions, or nullptr if an operation with this name doesn't exist. \param
+            /// operationName Name of operation \return Place representing operation
+            virtual Place::Ptr get_place_by_operation_name(const std::string& operationName);
+
+            /// \brief Returns an input port place by operation name and appropriate port index
+            /// \param operationName Name of operation
+            /// \param outputPortIndex Index of input port for this operation
+            /// \return Place representing input port of operation
+            virtual Place::Ptr
+                get_place_by_operation_and_input_port(const std::string& operationName,
+                                                      int inputPortIndex);
+
+            /// \brief Returns an output port place by operation name and appropriate port index
+            /// \param operationNameNname of operation
+            /// \param outputPortIndex Index of output port for this operation
+            /// \return Place representing output port of operation
+            virtual Place::Ptr
+                get_place_by_operation_and_output_port(const std::string& operationName,
+                                                       int outputPortIndex);
+
+            ///// Naming and annotation  /////
+
+            /// \brief Sets name for tensor. Overwrites existing names of this place
+            /// \param operation Tensor place
+            /// \param newName New name for this tensor
+            virtual void set_name_for_tensor(Place::Ptr tensor, const std::string& newName);
+
+            /// \brief Adds new name for tensor
+            /// \param operation Tensor place
+            /// \param newName New name to be added to this place
+            virtual void add_name_for_tensor(Place::Ptr tensor, const std::string& newName);
+
+            /// \brief Sets name for operation. Overwrites existing names of this place
+            /// \param operation Operation place
+            /// \param newName New name for this operation
+            virtual void set_name_for_operation(Place::Ptr operation, const std::string& newName);
+
+            /// \brief Unassign specified name from tensor place(s)
+            /// \param name Name of tensor
+            virtual void free_name_for_tensor(const std::string& name);
+
+            /// \brief Unassign specified name from operation place(s)
+            /// \param name Name of operation
+            virtual void free_name_for_operation(const std::string& name);
+
+            /// \brief Set name for a particular dimension of a place (e.g. batch dimension)
+            /// \param place Model's place
+            /// \param shapeDimIndex Dimension index
+            /// \param dimName Name to assign on this dimension
+            virtual void set_name_for_dimension(Place::Ptr place,
```

Comment:
I think we should have add_name_for_dimensions

Reply:
@slyalin do you agree add this method?

Reply:
Separate ticket #56772 is created for this